Set _NET_WM_WINDOW_TYPE to _NET_WM_WINDOW_TYPE_DIALOG.

de-vri-es opened the pull request:
This PR ensures that `_NET_WM_WINDOW_TYPE` is set to `_NET_WM_WINDOW_TYPE_DIALOG` so that window managers can act accordingly.

Most tiling window managers will tile windows without this hint, which is not really desirable. Fixes #247.

step- commented:
I wonder, given the list of "main dialogs", do you think `_NET_WM_WINDOW_TYPE_DIALOG` is the best choice for all of them? My take:

|   |   |   |
| - | - | - |
`--calendar`       | Display calendar dialog | Y
`--color`          | Display color selection dialog | Y
`--dnd`            | Display drag-and-drop box
`--entry`          | Display text entry dialog | Y
`--icons`          | Display box with application shortcut icons | ??
`--file`           | Display file selection dialog | Y
`--font`           | Display font selection dialog | Y
`--form`           | Display form dialog | ??
`--html`           | Display HTML dialog | N
`--list`           | Display list dialog | ??
`--multi-progress` | Display multi-progress bars dialog | Y
`--notebook`       | Display notebook dialog | ??
`--notification`   | Display notification icon | N
`--print`          | Display print dialog | Y
`--progress`       | Display progress indicator dialog | Y
`--text-info`      | Display text information dialog | N
`--scale`          | Display scale dialog | Y

de-vri-es commented:
I think most of those should still be dialogs. The main question for me is: should they be tiled or not by my tiling window manager. And since I'm always using `yad` as a tiny GUI for my terminal application, I always want them to float on top of the terminal. The notification icon is an exception, but there is no window anyway.

However, it might make sense to allow the user to override this by adding a `--no-dialog` option. Or a `--window-type dialog|splash|normal` option (and deprecate `--splash`). I do think the default should be to use the dialog window type.

step- commented:
> However, it might make sense to allow the user to override this by adding a `--no-dialog` option. Or a `--window-type dialog|splash|normal` option (and deprecate `--splash`). I do think the default should be to use the dialog window type.

I like the `--window-type` override. I think that for backwards-compatibility reasons the default should be the historical value, that is, `normal`. Then if you wanted to change the default for all dialogs in your machine because you use a tiling WM, you could `export YAD_OPTIONS="--window-type=dialog"` in your shell profile, or set this options in your dconf settings.

de-vri-es commented:
Fair. I implemented the `--window-type` option and added a deprecation warning for `--splash`.

I understand the reasoning for the default value, although I do think it only matters for tiling window managers, where the new default would be an improvement. Nevertheless, I'm just happy to have to option to make it a dialog window :)

de-vri-es commented:
@step- Do you think the changes are good like this?

step- commented:
@de-vri-es I didn't further comment your [commits](https://github.com/v1cont/yad/pull/248#issuecomment-1721022798) because (yad author) v1cont may at his discretion. As you probably know, he is in the military and has slowed down his rate of work on yad for obvious reasons.
I don't have write permissions on this repo. But I can look at your commits and comment here, if you still want me to.

de-vri-es commented:
Ah, I didn't know. But if you have the time and want to take a look, I'd still be thankful :)

step- commented:
I will, just give me a few days.

v1cont commented:
i don't want to change current default behavior. but i thing implement --window-type option is a good idea, so please look at e834dfd68feddf8736e8abfa8a5eb9389a7dcc75

de-vri-es commented:
This PR already preserves the default behavior :)
